File: mavkit-python/examples/connect_udp.py
import asyncio
import os

import mavkit


async def main():
    bind_addr = os.environ.get("MAVKIT_EXAMPLE_UDP_BIND", "0.0.0.0:14550")

    vehicle = await mavkit.Vehicle.connect_udp(bind_addr)
    identity = vehicle.identity()
    current_mode = vehicle.available_modes().current().latest()
    armed_sample = vehicle.telemetry().armed().latest()
    position_sample = await vehicle.telemetry().position().global_pos().wait()
    position = position_sample.value
    mode_name = current_mode.name if current_mode is not None else "unknown"
    armed = armed_sample.value if armed_sample is not None else None

    print(
        f"connected: sys={identity.system_id} comp={identity.component_id} "
        f"autopilot={identity.autopilot} vehicle_type={identity.vehicle_type} "
        f"mode={mode_name} armed={armed}"
    )

    print(
        f"telemetry: lat={position.latitude_deg:.7f} lon={position.longitude_deg:.7f}"
        f" alt_msl={position.altitude_msl_m:.1f} rel_alt={position.relative_alt_m:.1f}"
    )

    await vehicle.disconnect()


asyncio.run(main())

File: examples/connect_udp.rs
use mavkit::Vehicle;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let bind_addr =
        std::env::var("MAVKIT_EXAMPLE_UDP_BIND").unwrap_or_else(|_| "0.0.0.0:14550".to_string());

    let vehicle = Vehicle::connect_udp(&bind_addr).await?;
    let identity = vehicle.identity();
    let current_mode = vehicle.available_modes().current().wait().await?;
    let armed = vehicle.telemetry().armed().wait().await?.value;
    let position = vehicle.telemetry().position().global().wait().await?.value;

    println!(
        "connected: sys={} comp={} autopilot={:?} vehicle_type={:?} mode={} armed={}",
        identity.system_id,
        identity.component_id,
        identity.autopilot,
        identity.vehicle_type,
        current_mode.name,
        armed,
    );
    println!(
        "telemetry: lat={:.7} lon={:.7} alt_msl={:.1} rel_alt={:.1}",
        position.latitude_deg,
        position.longitude_deg,
        position.altitude_msl_m,
        position.relative_alt_m
    );

    vehicle.disconnect().await?;
    Ok(())
}
